packages/vuetify/types/index.d.ts

```diff
@@ -50,6 +47,48 @@ export interface Framework {
   rtl: boolean
 }
 
+export interface VuetifyPreset {
+  breakpoint: {
+    scrollBarWidth: Breakpoint['scrollBarWidth']
+    thresholds: Breakpoint['thresholds']
+  }
+  icons: {
+    iconfont: Icons['iconfont']
+    // TODO: Remove partial for v3
+    values: Partial<Icons['values']>
+  }
+  lang: {
+    current: Lang['current']
+    locales: Lang['locales']
+    t: Lang['t']
+  }
+  theme: {
+    dark: Theme['dark']
+    default: Theme['default']
+    disable: Theme['disable']
+    options: Theme['options']
+    themes: Theme['themes']
+  }
+}
+
+export interface UserVuetifyPreset extends GlobalVuetifyPreset {
+  preset?: GlobalVuetifyPreset
+}
+
+export interface GlobalVuetifyPreset {
```

Comment:
this should probably be called `VuetifyUseOptions` or something similar. I feel very uncertain of what a 'preset' actually is to us. To me it's a subset of the available VuetifyUseOptions that together can alter the look of vuetify. But VuetifyPreset also include things like `lang` which has nothing to do with the look.

Here's my suggestion
```ts
/** https://stackoverflow.com/questions/47914536/use-partial-in-nested-property-with-typescript */
type NestedPartial<T> = {
  [K in keyof T]?: T[K] extends Array<infer R> ? Array<NestedPartial<R>> : NestedPartial<T[K]>
}

export interface VuetifyPreset {
  breakpoint: Pick<Breakpoint, 'scrollBarWidth' | 'thresholds'>
  icons: Pick<Icons, 'iconfont' | 'values'>
  theme: Pick<Theme, 'dark' | 'default' | 'disable' | 'options' | 'themes'>
}

export interface UserVuetifyPreset extends NestedPartial<VuetifyPreset> {
  preset?: VuetifyPreset
}

export interface VuetifyUseOptions extends UserVuetifyPreset {
  components?: Dictionary<ComponentOrPack>
  directives?: Dictionary<Directive>
  transitions?: Dictionary<ComponentOrPack>
  lang?: Pick<Lang, 'current' | 'locales' | 't'>
  rtl?: boolean
}

```

Reply:
A preset is a collection of configuration options. I wrote tons of information regarding that in the documentation. We can change the types and the preset property is the only one that uses the type.